### PR TITLE
replace all `..` imports

### LINKED ISSUE #14021: Absolute imports
### Description

In PEP8 absolute imports are preferred over relative ones. I think we should switch, with a pre-commit hook, e.g. https://github.com/MarcoGorelli/absolufy-imports. We can do the whole repo in one go and add a git-blame-ignore if we want, though it's only imports being changed.

A nice explanatory block on the pros and cons of absolute imports from https://realpython.com/absolute-vs-relative-python-imports/#pros-and-cons-of-absolute-imports

> Pros and Cons of Absolute Imports
> Absolute imports are preferred because they are quite clear and straightforward. It is easy to tell exactly where the imported resource is, just by looking at the statement. Additionally, absolute imports remain valid even if the current location of the import statement changes. In fact, PEP 8 explicitly recommends absolute imports.

> Sometimes, however, absolute imports can get quite verbose, depending on the complexity of the directory structure. Imagine having a statement like this:

> from package1.subpackage2.subpackage3.subpackage4.module5 import function6
That’s ridiculous, right? Luckily, relative imports are a good alternative in such cases!

Absolute imports are better at preventing circular dependencies and are more explicit. Also, with `black` causing line wraps absolute imports are frequently equally verbose as the relative import.

### PR DESCRIPTION
**Pull request by nstarman**
Signed-off-by: Nathaniel Starkman (@nstarman) <nstarkman@protonmail.com>

### Description

Fixes #14021 

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: astropy/__init__.py [hunk 1]
```diff
@@ -255,11 +255,11 @@ def online_help(query):
         )
     ):
         # The last clause in the the above disjunction deserves explanation:
-        # When using relative imports like ``from .. import config``, the
+        # When using relative imports like ``from astropy import config``, the
```
Comment (mhvk) [inline astropy/__init__.py:258]: This change should not happen! (Or the whole comment needs adjusting)

File: astropy/units/format/utils.py [hunk 2]
```diff
@@ -7,10 +7,9 @@
 
 import warnings
 
+from astropy.units.utils import maybe_simple_fraction
```
Comment (mhvk) [inline astropy/units/format/utils.py:10]: Not a big deal, but the order here seems not ideal -- this is still an import within the submodule, so more "local" than `astropy.utils`.

**Review by mhvk — COMMENTED** [review]
Interesting to actually see the changes.  The `...` replacements are obviously better since they cross sub-module boundaries. Not all `..` are as obvious, mostly since in that case there was a clear link to the submodule. On the other hand, clearly we were not remotely consistent for `coordinates`, which is worse. I guess I'd prefer it if the imports had a "submodule-local" section. But that is separate from what is being tried here.

Anyway, the only real thing to do is to revert the change to the comment.

File: astropy/__init__.py (hunk 1 above)
Comment (nstarman) [reply astropy/__init__.py:258]: rolling it back.

**Review by mhvk — APPROVED** [review]
Looks all OK now! Thanks, @nstarman

**Review by dhomeier — APPROVED** [review]
LGTMT, thanks!

**Pull request merged**